includes/routes/user/attendance-mode.php

```diff
@@ -0,0 +1,64 @@
+<?php
+
+namespace Wporg\TranslationEvents\Routes\User;
+
+use Wporg\TranslationEvents\Attendee\Attendee;
+use Wporg\TranslationEvents\Attendee\Attendee_Repository;
+use Wporg\TranslationEvents\Event\Event_Repository_Interface;
+use Wporg\TranslationEvents\Routes\Route;
+use Wporg\TranslationEvents\Translation_Events;
+use Wporg\TranslationEvents\Urls;
+
+/**
+ * Toggle whether the current user is attending an event onsite or remotely.
+ * If the user is not currently marked as remote attendee, they will be marked as remote attendee.
+ * If the user is currently marked as as remote attendee, they will be marked as not remote attendee.
+ */
+class Attendance_Mode_Route extends Route {
```

Comment:
As a suggestion for a future PR, I think it could make sense to consolidate in a single route the functionality for managing attendance of a user to a given event.

There are now four routes that deal with attendance of a user to a given event:

- `/events/attend/$event_id`
- `/events/host/$event_id/$user_id`
- `/events/attendance-mode/$event_id/$user_id`
- `/events/$event_id/attendees/remove/$event_id`

We could replace the first three with a single route that creates or updates an `attendee`, e.g. `/events/$event_id/attendees/$user_id/set`, which could take the following parameters:

- `is_host`
- `is_remote`

This route can then be called both by the user (when clicking the attending buttons), and from an admin, when using the Manage Attendees page.


Additionally, we could change the structure of the existing `remove` route (the last one on the list above), so we end up with the following two routes:

- `/events/$event_id/attendees/$user_id/set`
- `/events/$event_id/attendees/$user_id/remove`

Reply:
I agree with this!